### DISCUSSION TITLE
Capacity Settings

### DISCUSSION THREAD

**Discussion by tcambell** [KG-Development/SmartBMSUtility#2] [Q&A] [answered]
Hi,

What a great app! 
I have a 10s lipo battery with 8000mAh capacity. How do I set/calibrate the total battery capacity and total cycle capacity so the capacity estimate indicator on the overview page is somewhat accurate? 

How is the capacity indicator determining capacity? Is it using voltage or keeping track of the coulombs used?

Thanks,
Tim

**Comment by NeariX67** [comment, marked as answer]
Total battery capacity is your 8000mAh that the battery was designed to have.
Cycle capacity is the actual capacity your battery has left. The BMS measures all the Ah that go through the BMS and adjusts the capacity accordingly. So do one or two FULL cycles on your battery to let your BMS calculate it for you.

**Reply by tcambell** [reply]
So I shouldn’t modify the total battery capacity and total cycle capacity settings on the configuration page? I’m assuming my BMS will adjust these settings after a few full charge and discharge cycles?

BTW what is you PayPal so I can send you som money for this great app

**Reply by NeariX67** [reply]
I've seem to be wrong here. [A nice documentation](https://raw.githubusercontent.com/FurTrader/OverkillSolarBMS/master/Overkill_Solar_BMS_Instruction_Manual.pdf) (see 3.2.1 and 3.2.2) from Overkillsolar has a nice peak into the bms functionality.

**Reply by sebastianleondelmoral** [reply]
Hi Justin! I bought SmartBMS Utility but I have an old white macbook. It has 10.11.6 last update and your app doesn`t work on it. Can you help me? Thanks!

**Reply by NeariX67** [reply]
Sorry, I can't. This version of MacOS is not compatible.

**Comment by NeariX67** [comment]
Hey, set your total battery capacity to 8000mAh and you are good to go. A few full charge and discharge cycle should provide the bms enough time to calibrate your battery capacity.

I would appreciate your donation of any amount! Thank you for your donation =) <3 [Paypal.me](https://www.paypal.com/paypalme/nearix846)

**Reply by tcambell** [reply]
Thanks,

data logging with an option to export the raw data for analysis would be awesome 
Notifications on the iPhone of any triggered protections would also be very helpful

**Comment by NeariX67** [comment]
Once i find some time i will get on it. Thank you very much for that 10 bucks =)

**Reply by tcambell** [reply]
Your welcome
Extra time for hobbies is a hard thing to find these days
I’ll keep checking for any updates. 
BTW. Im using your app to monitor a battery extender that I built for my ebike. Works great!

I’m always willing to donate more for updates

**Reply by NeariX67** [reply]
Are you using the GPS Features? As it's really cold outside rn, i haven't been able to take a ride. Everything working as expected and imperial units are displaying correctly?

**Reply by tcambell** [reply]
I haven’t tried the gps features yet. That’s next

**Reply by tcambell** [reply]
Everything appears to be functioning correctly in the app, including the imperial units. I’m really liking your app!

What information does the “Reset min/max values” reset? I couldn’t find a description for this feature and I’m afraid to use it!

Is there a way to disconnect the Bluetooth link from the app without having to go into the Bluetooth settings on the phone to unlink? It would be useful and convenient to unlink the Bluetooth link from the app. I think this could also save energy usage from the Bluetooth board when not in use. I noticed the led on the Bluetooth adapter stays lit until I turn off the Bluetooth on my phone.

I really like the GPS features, but noticed that the info is only instantaneous except for the distance. Additionally, the distance only records if the app is open. It would be nice if you could close the app and have it run in the background. I think it would also be useful to record and save a ride or use session (for whatever a person is using the app and BMS for like e-bike, e-skateboard etc.) This probably falls under data recoding which is indicated on your to do list.

During and after the recorded session the following information could be displayed and saved below on the GPS screen:
Ride or Session name
Date, start time, end time, elapsed time
Total distance 
Avg speed
Max speed
Battery start capacity (Wh)
Battery end capacity 
Consumed capacity 
Max Amps, Max power
Avg Amps, Avg power

please take all these comments as constructive. I’m very impressed by the app you have created and look forward to any features you decide to add.

thanks,

Tim

**Reply by NeariX67** [reply]
* The reset min/max values will reset values like "highest voltage", "lowest voltage", "peak power", "top speed", "peak amperage" to zero, you won't break anything hitting that :)
* Just swipe from the left edge to the right when you are connected, this will bring you to the devices menu and will disconnect you from the battery. You can do the same by hitting "< Devices" on the top left when you are connected. Btw: The app uses BLE which stands for Bluetooth Low Energy and I really hope that it deserves it's name. But disconnecting won't harm anyone either :D
* The main problem I see here is battery usage. This will consume A LOT of energy. Implementing this will take some time if I decide to do it, but i will need to a find a proper solution to fix unnecessary battery usage. Saving a ride seems like a pretty big task when it should include the route on a gps map, but your list is really all it needs and what could be easily implemented.

Maybe thats a thing for v1.3.0?
